students/perrymant/21-docker.md

```diff
@@ -0,0 +1,99 @@
+# Docker
+
+> Docker runs processes in isolated containers. A container is a process which runs on a host. The host may be local or remote. When an operator executes docker run, the container process that runs is isolated in that it has its own file system, its own networking, and its own isolated process tree separate from the host. - [Docker](https://docs.docker.com/)
+
+- What is a container? at its simplest it's a sandbox for a process - allowing the process to have its own namespace, restricting certain capabilities and imposing resource limits as required.
+- Container technology allows multiple isolated user space instances to be run on a single host.
+- Containers are generally considered a lightweight or lean technology because they require limited overhead, as each process can be run on very small Linux distributions.
+- Docker encourages microservices architecture.
+- Images are the building blocks of the Docker world. You launch your containers from images.
+- `image` vs `container`: an image is akin to an OOP class (the blueprint), and the container is akin to an OOP object (the instance).
+- Docker stores the images you build in registries. There are two types of registries: public and private, and can be found on (Docker Hub)[https://hub.docker.com], which contains over 10,000 images that other people have built and shared.
+- A Docker container comprises:
+    - An image format.
+    - A set of standard operations.
+    - An execution environment.
+
+### Overview of common commands
```

Comment:
Yeah, I understand what you say about the quick overview, just to keep an eye in the order of in depth explanations to also be easily to you relate them when you be looking for it. And yes, I'm in agree about you say to put Mongo specific Docker tasks in the corresponding notes. By the way, I think is very good the way you are naming your notes files because this is very clear to make a quick look at what was sawn in mentory session. I think the way I'm doing I have not a quick idea what I saw. I think maybe one day I will take your idea and rename my notes to some format similar yours ;)